Mathlib/GroupTheory/Perm/Cycle/Basic.lean

```diff
@@ -985,3 +987,186 @@ theorem sum_mul_sum_eq_sum_perm (hσ : σ.IsCycleOn s) (f g : ι → α) :
   sum_smul_sum_eq_sum_perm hσ f g
 
 end Finset
+
+namespace Equiv.Perm
+
+theorem subtypePerm_apply_pow_of_mem {g : Perm α} {s : Finset α}
+    (hs : ∀ x : α, x ∈ s ↔ g x ∈ s) {n : ℕ} {x : α} (hx : x ∈ s) :
+    ((g.subtypePerm hs ^ n) (⟨x, hx⟩ : s) : α) = (g ^ n) x := by
+  simp only [subtypePerm_pow, subtypePerm_apply]
+
+theorem subtypePerm_apply_zpow_of_mem {g : Perm α} {s : Finset α}
+    (hs : ∀ x : α, x ∈ s ↔ g x ∈ s) {i : ℤ} {x : α} (hx : x ∈ s) :
+    ((g.subtypePerm hs ^ i) (⟨x, hx⟩ : s) : α) = (g ^ i) x := by
+  simp only [subtypePerm_zpow, subtypePerm_apply]
+
+variable [Fintype α] [DecidableEq α]
+
+/-- Restrict a permutation to its support -/
+def subtypePermOfSupport (c : Perm α) : Perm c.support :=
+  subtypePerm c fun _ : α => apply_mem_support.symm
+
+/-- Restrict a permutation to a Finset containing its support -/
+def subtypePerm_of_support_le (c : Perm α) {s : Finset α}
+    (hcs : c.support ⊆ s) : Equiv.Perm s :=
+  subtypePerm c (isInvariant_of_support_le hcs)
+
+/- Four lemmas added but maybe for nothing worth -/
```

Comment:
In that case I would leave them out of this PR, but save them somewhere. I guess they will remain in the big PR that depends on this one.

Reply:
I can very well do that. In fact, I reworked largely the subsequent PR on branch#ACL/ConjClassCount and certainly many such lemmas are not used anymore.

Reply:
I deleted those lemmas.